### PR TITLE
Optimization for shell sort

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](hxtps://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by cclauss** [comment]
I do not quite understand the title of this PR.  In what way is the current implementation broken?

**Comment by shellhub** [comment]
> I do not quite understand the title of this PR. In what way is the current implementation broken?

We should't insert value instead of swapping elements of array.

**Comment by mateuszz0000** [comment]
Is your implementation improve performance or what? If so please write proper tests for proof that your solution is better than already exists.

**Review by mateuszz0000 — CHANGES_REQUESTED** [review]
Need additional tests for performance

**Comment by poyea** [comment]
Maybe we can strictly follow this?: https://en.wikipedia.org/wiki/Shellsort#Pseudocode 
Sometimes it's just a matter of styles.

**Review by poyea — CHANGES_REQUESTED** [review]
We may add a reference to the wiki page as well.

**Comment by shellhub** [comment]
@poyea Done. Please review again.

File: sorts/shell_sort.py [hunk 1]
```diff
@@ -1,13 +1,5 @@
...
-This is a pure Python implementation of the shell sort algorithm
-
-For doctests run following command:
-python -m doctest -v shell_sort.py
-or
-python3 -m doctest -v shell_sort.py
-
-For manual testing run:
-python shell_sort.py
+https://en.wikipedia.org/wiki/Shellsort#Pseudocode
```
Comment (poyea) [inline sorts/shell_sort.py:2]: I would leave the first line like this
```suggestion
"""
This is a Python implementation of the shell sort algorithm.

Reference: https://en.wikipedia.org/wiki/Shellsort#Pseudocode
```

**Review by poyea — APPROVED** [review]
I would approve this as the original version has really no reference to the algorithm (or at least, the implementation didn't strictly follow the algorithm). Thank you for your pull request!🤩

File: sorts/shell_sort.py [hunk 2]
```diff
@@ -25,15 +17,16 @@ def shell_sort(collection):
...
+
     gaps = [701, 301, 132, 57, 23, 10, 4, 1]
     for gap in gaps:
         for i in range(gap, len(collection)):
             insert_value = collection[i]
-            j = i - gap
-            while j >= 0 and insert_value < collection[j]:
-                collection[j + gap] = collection[j]
+            j = i
+            while j >= gap and insert_value < collection[j - gap]:
```
Comment (poyea) [inline sorts/shell_sort.py:26]:
```suggestion
            while j >= gap and collection[j - gap] > insert_value:
```
for readability

**Comment by shellhub** [comment]
@poyea Done

**Comment by shellhub** [comment]
@mateuszz0000 Can you review again. Thanks :)

**Review by mateuszz0000 — APPROVED** [review]

**Pull request merged**